src/main/java/com/google/api/codegen/util/SampleConfigSanitizer.java

```diff
@@ -0,0 +1,62 @@
+/* Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.api.codegen.util;
+
+import java.io.FileReader;
+import java.io.IOException;
+import java.io.Reader;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import org.yaml.snakeyaml.Yaml;
+
+public class SampleConfigSanitizer {
```

Comment:
I originally went with `Reader`, but decided to change the name because this class does not really do the yaml reading (`MultiYamlReader` still does that, though it's not doing quite good of a job but that's still the truth for now). This class just check the input and filter out whatever is not recognized as a sample config file, thus the name `Sanitizer`. 

I want to ultimately get rid of `MultiYamlReader` for samples and use this class to read the sample configs, and probably we can rename the class when this is done.

Reply:
Even if we keep `MultiYamlReader` (I'm curious why you want to replace it), I think `Sanitizer` is not an informative name. I guess you're right in that this returns the file names, not contents, so maybe something with "Filter" in the name. But whatevs, it's a nit we can worry about separately.

Reply:
Because `MultiYamlReader` uses `YamlReader` from `api-compiler`, which doesn't support reading yamls with anchors.

Reply:
Excellent reason!